crates/ruma-common/src/events/room/message/relation_serde.rs

```diff
@@ -18,8 +18,20 @@ impl<'de> Deserialize<'de> for Relation {
         let ev = EventWithRelatesToJsonRepr::deserialize(deserializer)?;
 
         #[cfg(feature = "unstable-msc3440")]
-        if let Some(RelationJsonRepr::Thread(ThreadJsonRepr { event_id, is_falling_back })) =
-            ev.relates_to.relation
+        if let Some(RelationJsonRepr::ThreadStable(ThreadStableJsonRepr {
```

Comment:
Same here, branches can be merged because they have the same logic and the same captured variables in the pattern.